### ISSUE TITLE
rate limit issue

### ISSUE THREAD

**Issue by madsonic** [nus-cs2103/cs2103-pr-bot#57]
When there are many PRs to process, the bot usually hits a rate limit imposed by github. although the quota is not exhausted, too many requests are sent in a short time causing the bot to be halted for a short period.

**Comment by damithc** [comment]
This happens only at the first time the bot is activated (because of the backlog), am I right?  If that is the case, the solution is to activate the bot before PRs start coming in.

**Comment by madsonic** [comment]
yes backlog situation would result in hitting the limit. However there are certain reasons that could discourage the bot from being deployed e.g. team info not complete, bot account has insufficient permissions. 
Either improve on how the bot reads in team info (while filtering out incomplete one otherwise we run into #58) or fixing the permissions (this is relatively easy and controllable). While we could take steps to solve each of this, i find that having direct control measures on the rate limit would have a more robust outcome.

**Comment by damithc** [comment]
What's the impact of hitting the rate limit? If it simply delays the processing of PRs (i.e., the backlog clears itself automatically over a few hours), it's something we can live with. But if the impact is bigger, we can consider a more permanent solution. GitHub tends to change their behavior frequently. The less code we write against GitHub, the less chance of things breaking. Also, the more complex the code, the harder it is to find tutors to take over the bot in future. That's why we should try to keep the bot as simple as possible. :-)

**Comment by madsonic** [comment]
i personally feel that it is bad developer experience. He has to re-run it multiple times to achieve the same effect if he actually know how to solve it. Newcomers would not really understand what is happening without extra documentation (more complex documentation). But as the maintainer if you think that this creates more complexity and is unnecessary, you can close this issue and I will document this step

**Comment by damithc** [comment]
If it requires the dev to manually rerun things, that's a good reason to look for a better way. I assumed the bot will automatically rerun periodically and the backlog will clear itself automatically over time.

You can make a decision on which is easier for future devs.  It's hard for me to say which option is harder because I haven't experienced either option.
